enhancements/storage/csi-efs-operator.md

```diff
@@ -0,0 +1,341 @@
+   * Most of the operator just installs / removes CSI driver, i.e. manages driver's RBAC rules, CredentialsRequest,
+     DaemonSet, Deployment, CSIDriver and StorageClasses. See existing
+     [AWS EBS operator](https://github.com/openshift/aws-ebs-csi-driver-operator/tree/master/assets) for objects it
+     manages, EFS should be very similar in this regard.
+   * `SharedVolume` processing on top of it is very trivial, it just creates PV + PVC.
+
+5. Ship [efs-utils](https://github.com/aws/efs-utils) as a base image for the EFS CSI driver image.
+
+6. Update installer to delete Access Points and EFS volumes that have `owner` tag when destroying a cluster.
+   * Existence of EFS volume using a VPC / Subnets actively prevents installer to delete them.
+     `openshift-install destroy cluster` gets stuck at VPC (or subnet?) deletion, because AWS does not allow to delete
+     a network that's used.
+   * This way, cluster admins can tag the volumes to be deleted when the cluster is deleted (this is opt in!).
+     At least CI should use this approach to ensure there are no leftovers.
+
+### Implementation Details/Notes/Constraints [optional]
+
+### Risks and Mitigations
+
+Currently, all cloud-based CSI drivers shipped by OCP are installed when a cluster is installed on the particular cloud.
+AWS EFS will not be installed by default. We will include a note about opt-in installation in docs.
+
+After installation of the new operators, either as a fresh install or upgrading the existing one, users must create the
+CR when the operator is installed on a new cluster
+([example of CR for AWS EBS](https://github.com/openshift/cluster-storage-operator/blob/master/assets/csidriveroperators/aws-ebs/10_cr.yaml),
+EFS will be very similar). This is something that was not necessary in the previous releases of
+the operator. We will add a release note and introduce an (info) alert to remind admins to create the CR.
+
+## Design Details
+
+### Rewrite to library-go & ClusterCSIDriver CRD
+
+This should be pretty straightforward. We already ship number of CSI drivers that use library-go. Example usage:
+[AWS EBS CSI driver](https://github.com/openshift/aws-ebs-csi-driver-operator/blob/383a7638b37a4b9a9831c7747e8c499eedcf030f/pkg/operator/starter.go#L67).
+EFS CSI driver operator should not be much more complicated.
+
+Library-go does not support un-installation of CSI driver (and static resources created by `StaticResourcesController`),
+this must be implemented there.
+
+Note: When the *operator* is un-installed, nothing happens to the CRD, CR, the installed CSI driver, existing PVs and
+PVCs. Everything is working, just not managed by the operator. When the *operand* is un-instaled by removing the
+operator CR, the driver will be removed. Existing PVs, PVCs and SharedVolumes will be still present in the API server,
+however, they won't be really usable. Nobody can mount / unmount EFS volumes without the driver.
```

Comment:
I have a question here, if the `operator` is un-installed, when we delete the operator CR, the efs csi driver will not be removed? IIUC, the `operator` will watch the CR and process the driver install or un-install logic.